### ISSUE TITLE
The inequalities should auto-evaluate in some cases.

### ISSUE THREAD

**Issue by ShubhamKJha** [sympy/sympy#16582]
Consider the following code.
```python
In [1]: from sympy import *

In [2]: x = symbols('x', real=True)

In [3]: x<-oo
Out[3]: x < -oo

In [4]: x > oo
Out[4]: x > oo
```
Since no real numbers should be greater than `oo` or less than `-oo`, the result should have been `False`.

**Comment by sylee957** [comment]
Actually
```python
from sympy import *

x = symbols('x', real=True, finite=True)
x > oo
```

It returns `False`

**Comment by oscargus** [comment]
On the other hand:
```
In [1]: from sympy import oo

In [2]: oo>oo
Out[2]: False
```
so it seems to make sense that one should be able to evaluate these directly.

There is the usual discussion about complex numbers though... But it should be enough that `x` is real to return `False` for `x > oo` etc.￼

**Comment by ShubhamKJha** [comment]
Hi @sylee957 `real` can be either `finite` or `infinite`.In both cases, the above results holds. But it appears the current master doesn't evaluate it.
```python
In [6]: from sympy import symbols

In [7]: x = symbols('x', real=True, infinite=True)

In [8]: x > oo  # since any real number would be less than or equal to oo
Out[8]: x > oo  # The result should've been False
```

**Comment by sylee957** [comment]
I see 
```python
x = symbols('x', real=True, positive=True)
x > oo
```
and
```python
x = symbols('x', real=True, positive=True)
x >= oo
```
both returning `False`
Though `positive` shouldn't mean `finite`

**Comment by ShubhamKJha** [comment]
Apparently, `real  ==  positive | negative | zero` is in `_assume_rules` in `assumptions.py` and consider the following:
```python
>>> x,y,z = Dummy(positive=True),Dummy(negative=True),Dummy(zero=True)
>>> x>oo,y>oo,z>oo
(False, False, False)
```
Hence, it should be enough that x is real to return False for x > oo

**Comment by oscarbenjamin** [comment]
Currently real means extended real in the assumptions system. This means that `oo` is a possible value for a symbol `x = Symbol('x', real=True)`. So unless that is changed/fixed then `x > oo` would have to be `None`.

**Comment by ShubhamKJha** [comment]
> So unless that is changed/fixed then `x > oo` would have to be `None`.

@oscarbenjamin `oo > oo` returns `False` in SymPy.

**Comment by oscarbenjamin** [comment]
> > So unless that is changed/fixed then `x > oo` would have to be `None`.
> 
> @oscarbenjamin `oo > oo` returns `False` in SymPy.

It should give None rather than False.

**Comment by sylee957** [comment]
Actually, the I see comparisons of sympy objects other than `==`, are using `BooleanTrue` and `BooleanFalse`, rather than python's default ones.
And there is currently nothing equivalent to `None` in sympy boolean algebra types, instead it uses unevaluated relational class, so there can be some consistency issue if we force it to be `None`

**Comment by oscarbenjamin** [comment]
Oh good point. Maybe it should remain unevaluated:
```julia
In [72]: Gt(oo, oo, evaluate=False)                                                                                                            
Out[72]: ∞ > ∞
```

**Comment by jksuom** [comment]
The relation `x > y` should be equivalent to `x >= y and x != y` so it is false if `x == y` is true.

**Comment by oscarbenjamin** [comment]
Maybe that does make sense. I hadn't really interpreted SymPy's `oo == oo` as meaning that the objects are mathematically (rather than structurally) equal. I guess they are considered mathematically equal in the extended reals.

Going back to the original issue we have
```julia
In [101]: xr = Symbol('xr', real=True)                                                                                                         

In [102]: xrf = Symbol('xrf', real=True, finite=True)                                                                                          

In [103]: xr < oo                                                                                                                              
Out[103]: xr < ∞

In [104]: xrf < oo                                                                                                                             
Out[104]: True
```
So the crux of it is that currently real does not imply finite.

I do think that should be fixed but is there already an open issue?

On the other hand for positive we have
```julia
In [105]: xp = Symbol('xp', positive=True)                                                                                                     

In [106]: xpf = Symbol('xp', positive=True, finite=True)                                                                                       

In [107]: xp < oo                                                                                                                              
Out[107]: True

In [108]: xpf < oo                                                                                                                             
Out[108]: True
```
this is incorrect since positive does not imply finite:
[https://github.com/sympy/sympy/blob/1b7514db3923012f6f0aad14315b24e95bef639c/sympy/core/assumptions.py#L182](https://github.com/sympy/sympy/blob/1b7514db3923012f6f0aad14315b24e95bef639c/sympy/core/assumptions.py#L182)

**Comment by ShubhamKJha** [comment]
> It should give None rather than False.

I don't think the result should be 'None' since `no infinity is greater than any other infinity` holds, and hence the `oo > oo` should result in `False`.
Python follows the same `math.inf > math.inf => False`. Even when I checked on WolframAlpha, the result is `False` with `inf > inf`.

> this is incorrect since positive does not imply finite:

This appears another issue to be handled.

**Comment by oscarbenjamin** [comment]
> I don't think the result should be 'None' since `no infinity is greater than any other infinity` holds, and hence the `oo > oo` should result in `False`.

This depends on your interpretation of infinity. I would view an expression involving infinity in terms of limits in some way. So when I see `oo > oo` I think of each object representing some expression that grows without bound e.g. `(1/x**2) > (1/x)` as x goes to zero. In that case the lhs is greater than the rhs but the reverse can also be true. In the limit both expressions "become infinity" but I wouldn't say that they become equal since even their ratio can be infinite.

**Comment by asmeurer** [comment]
There's been a lot of discussions about whether the real assumption should include infinity. See https://github.com/sympy/sympy/pull/2538#issuecomment-26742610. 

As for oo < oo, I think I it should return False as it does now.

**Comment by oscargus** [comment]
> As for oo < oo, I think I it should return False as it does now.

This also means that `x > oo` is False even if `x` in `ExtendedReal`. (Or are there arguments against that?)

**Comment by oscargus** [comment]
I can say that I sympathize with the limit/asymptotic argument. But when reaching `oo` I would still assume that it is the same value. Hence, the error in both
```
In [6]: Limit(1/x, x, 0).doit() > Limit(1/x**2, x, 0).doit()
Out[6]: False

In [7]: Limit(1/x, x, 0).doit() < Limit(1/x**2, x, 0).doit()
Out[7]: False

```
returning the same answer is not the comparison but how the comparison is formulated.

**Comment by sylee957** [comment]
I think that in the extended real algebra, directly evaluating the divergent limit elementwise, and doing the comparison should be avoided.

However, I see the relational also automatically evaluates `doit` to its elements, 

![image](https://user-images.githubusercontent.com/34944973/55714210-2f978c80-5a2d-11e9-8c61-0bb0fe5c396b.png)

So, in this case, if any divergent limit is found, should they first be joined into a single limit, and be compared with zero?

**Comment by jksuom** [comment]
> This also means that x > oo is False even if x in ExtendedReal.

That is true. `oo` is the greatest ExtendedReal. The set of extended real numbers is the *order completion* the ordered set of rational numbers. It the the smallest totally ordered set containing the rationals where each subset has a least upper bound and a greatest lower bound. (Its structure is defined completely in terms of order. The arithmetic laws of the rational numbers can be extended to the completion but only partially.)

**Comment by ShubhamKJha** [comment]
Closed in #16956
